pkg/terraform/terraform.go

```diff
@@ -23,27 +24,49 @@ const (
 
 	// VarFileName is the default name for Terraform var file.
 	VarFileName string = "terraform.tfvars"
+
+	// TargetBootstrap is the default name for the bootstrap directory
+	TargetBootstrap string = "bootstrap"
+
+	// TargetCompat is the name for the legacy terraform directory
+	TargetCompat string = "compat"
 )
 
+// GetStateFileName returns the name of the terraform state file. If no target
+// is specified or compat is specified, default to the old state file name.
+func GetStateFileName(target string) string {
+	if target != "" && target != TargetCompat && !LegacyMode() {
+		return fmt.Sprintf("terraform.%s.tfstate", target)
+	}
+
+	return StateFileName
```

Comment:
nit: The new format is the happy scenario. The old format is the exceptional case. This will also make it cleaner when it comes time to remove the old format.
```suggestion
    	if target == "" || target == "compat" {
    	   	return StateFileName
    	}
    	return fmt.Sprintf("terraform.%s.tfstate", target)
```
Although, I suggest that we can also just get rid of the old format entirely and use `terraform.compat.tfstate`.

Reply:
I've gone with your second recommendation here. The original goal was to keep the code compatible with the way the original terraform worked. This makes things easier.